app/assets/javascripts/discourse/tests/integration/components/search-menu-test.gjs

```diff
@@ -12,6 +16,20 @@ module("Integration | Component | search-menu", function (hooks) {
   setupRenderingTest(hooks);
 
   test("rendering standalone", async function (assert) {
+    pretender.get("/search/query", (request) => {
+      if (request.queryParams.type_filter === DEFAULT_TYPE_FILTER) {
+        // posts/topics are not present in the payload by default
```

Comment:
This comment here makes me think that this is a good candidate for a system test. A test that does essentially what the video in this PR does would cover this well, and we may not need some of the JS unit tests that rely on stubs.

Reply:
I would prefer to delay a larger test refactor until after we have ripped out the widget implementation. It will be much easier to dissect the tests then.

Reply:
Doesn't have to be part of this PR. And I don't think we need a large refactor, but adding 1-2 system tests for the search behaviours in a follow-up PR would be good.